### PR TITLE
fix: Stop crashing when redis is offline and ignoreErrors on cache op…

### PR DESCRIPTION
**Pull request by mesinger**
### Description of change
PR #7630 introduced the ```ignoreErrors``` flag for ```ConnectionOptions```. For me this updated did not fix my problem (also mentioned in the same PR #7630).
From my subjective opinion, typeorm should still function even though redis caching is not available. typeorm will be able to query like regular from the database overall increasing the resilience of systems using typeorm.

This fix adds an ```error``` callback for the internally used redis client, logging instead of throwing an error when ```ignoreErrors``` is set.

### Pull-Request Checklist

- [x] Code is up-to-date with the `master` branch
- [x] `npm run lint` passes with this change
- [x] `npm run test` passes with this change
- [ ] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change
- [ ] Documentation has been updated to reflect this change
- [x] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/typeorm/typeorm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: src/cache/RedisQueryResultCache.ts [hunk 1]
```diff
@@ -54,6 +54,13 @@ export class RedisQueryResultCache implements QueryResultCache {
                 ...cacheOptions?.options,
                 legacyMode: true
             });
+            this.client.on("error", (err: any) => {
```
Comment (Ginden) [inline src/cache/RedisQueryResultCache.ts:57]: Wouldn't it be better to do:

```ts
if (typeof this.connection.options.cache === "object" && this.connection.options.cache.ignoreErrors) {
  this.client.on("error", (err: any) => {
    this.connection.logger.log("warn", err);
  });
```

**Comment by pleerock** [comment]
Thank you for contribution! 🎉

**Comment by Juantrix123** [comment]
This PR introduces an error in RedisQueryResultCache class, since the connect method of the redis client is not called.
This causes (ClientClosedError: The client is closed) when cache is used

**Comment by mesinger** [comment]
You are right, should be fixed here https://github.com/typeorm/typeorm/pull/8894

**Pull request merged**